packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2406,17 +2479,18 @@ export class ContainerRuntime
 				this.blobManager.processBlobAttachOp(messageWithContext.message, local);
 				break;
 			case ContainerMessageType.IdAllocation:
-				assert(
-					this.idCompressor !== undefined,
-					0x67c /* IdCompressor should be defined if enabled */,
-				);
-
 				// Don't re-finalize the range if we're processing a "savedOp" in
 				// stashed ops flow. The compressor is stashed with these ops already processed.
 				if (
+					!this.skipSavedCompressorOps ||
```

Comment:
I am not clear on what the interaction between `this.skipSavedCompressorOps` and the other condition regarding `savedOp` is.
With the `!this.skipSavedCompressorOps` check added, if there is no pendingIdCompressorState on load, this check will be true and the op will be processed even if the metadata says `savedOp`? Is this scenario not possible anymore?

In any case, can you please update the comment above to incorporate the new condition.

Reply:
I've talked to Taylor about it if there is a way to simplify it. Basically ID compressor does not behave as DDS, where having stashed ops is sufficient to rehydrate state. Instead, it serializes final state, and thus skips all the ops (stashed or not) that precede that state (savedOp means - it's an op that precede final sequence number of serialized container).
I'm updating comment to make it easier to follow, but overall observation - logic around ID compressor is not great, and I do not see an easy way to make it simpler.

Reply:
Okay. The comment helps for sure.